### PR TITLE
Reinstated Logger calls in Core

### PR DESCRIPTION
**Pull request by hughwin**
I have an email from Richard, he says it's okay as it's a debugging feature!

### CODE REVIEW

**Comment by rjs68** [comment]
image
Can we move where "Game Started" is printed from please? It's currently inaccurate where it is

File: src/main/java/com/toptrumps/cli/CommandLineController.java [hunk 1]
```diff
@@ -91,13 +91,14 @@ private void startNewGame(int numberOfOpponents) {
         while (players.size() != 1) {
             // == GAME START UP ==
             roundNumber++;
+            Logger.getInstance().logToFileIfEnabled(roundNumber + "");
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineController.java]: Can you add some text to show that this is the round number? At the moment it just logs the number and it's not clear what this is

File: src/main/java/com/toptrumps/cli/CommandLineController.java [hunk 2]
```diff
@@ -91,13 +91,14 @@ private void startNewGame(int numberOfOpponents) {
...
             // Retrieves the human player;
             humanPlayer = getHumanPlayer(players);
             isHumanAlive = humanPlayer != null;
             // Checks if the humanPlayer has not been eliminated
             if (isHumanAlive) {
                 view.showRoundStart(activePlayer, humanPlayer, roundNumber, communalPile.size());
-            }
+            };
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineController.java]: Why have you added a semi colon here? It's an if block - it doesn't need one

**Comment by hughwin** [comment]
> 
> 
> image
> Can we move where "Game Started" is printed from please? It's currently inaccurate where it is

I've taken it completely as it wasn't required.

**Review by rjs68 — CHANGES_REQUESTED** [review]
It's looking much better now - just a few things to look at!
I tried to look for these remaining requirements but I couldn't find them in the log file? Can you confirm they're included please? I think for the second it wants us to print all the users values for that category.
• The contents of the communal pile when cards are added or removed from it
• The category selected and corresponding values when a user or computer selects a category

File: src/main/java/com/toptrumps/cli/CommandLineView.java [hunk 3]
```diff
@@ -193,7 +194,10 @@ public void showNoNumberSelected() {
 
     public void showSelectedAttribute(String playerName, String selectedAttribute, boolean skipPrintAnimation) {
         String message = String.format("%s selected the attribute %s", playerName, selectedAttribute);
-        typePrint(SLOW_SPEED, message, skipPrintAnimation);
+      
+        Logger.getInstance().logToFileIfEnabled(message);
+        typePrint(SLOW_SPEED, message);
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineView.java]: This method needs to be passed the skipPrintAnimation variable

**Review by rjs68 — CHANGES_REQUESTED** [review]
1 more minor thing - I guess it came about from the merge conflicts!

File: src/main/java/com/toptrumps/cli/CommandLineView.java [hunk 4]
```diff
@@ -24,8 +24,9 @@
 
     public void showWelcomeMessage() {
         printWelcomeBanner();
-        typePrint(SLOW_SPEED, "\nTo start a new game, enter f \nTo see game statistics, enter s \nTo quit, enter q", false);
-        Logger.getInstance().logToFileIfEnabled("Game started");
+
+        typePrint(SLOW_SPEED, "\nTo start a new game, enter f \nTo see game statistics, enter s \nTo quit, enter q");
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineView.java]: I missed this one - please add false as the final argument to typePrint

**Review by rjs68 — CHANGES_REQUESTED** [review]

File: src/main/java/com/toptrumps/cli/CommandLineView.java (hunk 4 above)
Comment (hughwin) [reply src/main/java/com/toptrumps/cli/CommandLineView.java]: Done. I hate merges

File: src/main/java/com/toptrumps/cli/CommandLineView.java [hunk 5]
```diff
@@ -24,8 +24,9 @@
 
     public void showWelcomeMessage() {
         printWelcomeBanner();
-        typePrint(SLOW_SPEED, "\nTo start a new game, enter f \nTo see game statistics, enter s \nTo quit, enter q", false);
-        Logger.getInstance().logToFileIfEnabled("Game started");
+
+    typePrint(SLOW_SPEED, "\nTo start a new game, enter f \nTo see game statistics, enter s \nTo quit, enter q", false);
```
Comment (rjs68) [inline src/main/java/com/toptrumps/cli/CommandLineView.java]: This should be indented

**Review by rjs68 — CHANGES_REQUESTED** [review]
image
The selected attribute statement is duplicated, please remove one of them - preferably the one that prints the attribute value as well

**Review by rjs68 — APPROVED** [review]

**Review by nathanieldschneider — APPROVED** [review]
yeet

**Pull request merged**